File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/OpenMeteoWeatherApi.java
package wangdaye.com.geometricweather.weather.apis;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import wangdaye.com.geometricweather.weather.json.openmeteo.OpenMeteoWeatherResult;

/**
 * Open-Meteo API
 */
public interface OpenMeteoWeatherApi {

    // TODO: Check that I remembered correctly that we need this windspeed_unit
    @GET("v1/forecast?windspeed_unit=ms&timezone=auto&timeformat=unixtime")
    Observable<OpenMeteoWeatherResult> getWeather(@Query("latitude") double latitude,
                                                  @Query("longitude") double longitude,
                                                  @Query("daily") String daily,
                                                  @Query("hourly") String hourly,
                                                  @Query("forecast_days") int forecastDays,
                                                  @Query("past_days") int pastDays,
                                                  @Query("current_weather") boolean currentWeather);
}

File: app/src/main/java/org/breezyweather/weather/openmeteo/OpenMeteoWeatherApi.kt
package org.breezyweather.weather.openmeteo

import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import org.breezyweather.weather.openmeteo.json.OpenMeteoWeatherResult

/**
 * Open-Meteo API
 */
interface OpenMeteoWeatherApi {
    @GET("v1/forecast?timezone=auto&timeformat=unixtime")
    fun getWeather(
        @Query("latitude") latitude: Double,
        @Query("longitude") longitude: Double,
        @Query("daily") daily: String,
        @Query("hourly") hourly: String,
        @Query("forecast_days") forecastDays: Int,
        @Query("past_days") pastDays: Int,
        @Query("current_weather") currentWeather: Boolean
    ): Observable<OpenMeteoWeatherResult>
}